src/config.ts

```diff
@@ -36,3 +45,50 @@ export const PARTITIONS_CONSUMED_CONCURRENTLY = process.env
 export const PORT = Number(process.env.PORT || 7000);
 
 export const ENABLED = process.env.ENABLED === "true";
+
+/**
+ * VAULT configurations
+ */
+export const USE_VAULT = process.env.USE_VAULT === "true";
+export const VAULT_TOKEN = process.env.VAULT_TOKEN as string;
+export const VAULT_AUTH_METHOD = process.env.VAULT_AUTH_METHOD as
+  | "token"
+  | "kubernetes";
+export const VAULT_URL = process.env.VAULT_URL as string;
+export const VAULT_ROLE = process.env.VAULT_ROLE as string;
+export const VAULT_K8_TOKEN_PATH = process.env.VAULT_K8_TOKEN_PATH as string;
+export const VAULT_MONGO_SECRET_PATH = process.env
+  .VAULT_MONGO_SECRET_PATH as string;
+const REQUIRED_VAULT_CONFIGS = {
+  VAULT_AUTH_METHOD,
+  VAULT_URL,
+  VAULT_ROLE,
+  VAULT_MONGO_SECRET_PATH
+};
+const missingValue = ([key, value]: [string, any]) => !value;
+if (USE_VAULT && Object.entries(REQUIRED_VAULT_CONFIGS).some(missingValue)) {
```

Comment:
maybe in the future we should consider a bootstrap before the app is listening, so we can do these checks and run some setup stuff. We could use clinical as an example.

Reply:
this isn't a request just a suggestion

Reply:
these checks will fail the app before anything can start up so it'll have the same effect as a bootstrap so I don't think we really need one. can keep here for simplicity

Reply:
yea, that's cool with me :+1: